js/core/src/dependency.ts

```diff
@@ -364,7 +369,7 @@ export class DependencyManager {
 					if (dep_submission.type === "void") {
 						unset_args.forEach((fn) => fn());
 					} else if (dep_submission.type === "data") {
-						this.handle_data(dep.outputs, dep_submission.data);
+						await this.handle_data(dep.outputs, dep_submission.data);
```

Comment:
added these awaits because otherwise, if there was a .then that triggered another function, and it used the outputs of the first dependency as inputs, the gathered data would not reflect the updated data sometimes, annoying race condition